### PR TITLE
[.NET SDK] Add Luis threshold

### PR DESCRIPTION
**Pull request by JasonSowers**

### CODE REVIEW

File: CSharp/Library/Microsoft.Bot.Builder/Luis/LuisModel.cs [hunk 1]
```diff
@@ -167,14 +178,16 @@ public static Uri UriFor(LuisApiVersion apiVersion, string domain = null)
         /// <param name="subscriptionKey">The LUIS subscription key.</param>
         /// <param name="apiVersion">The LUIS API version.</param>
         /// <param name="domain">Domain where LUIS model is located.</param>
-        public LuisModelAttribute(string modelID, string subscriptionKey,
-            LuisApiVersion apiVersion = LuisApiVersion.V2, string domain = null)
+	/// <param name="threshold">Threshold for the top scoring intent.</param>
```
Comment (willportnoy) [inline CSharp/Library/Microsoft.Bot.Builder/Luis/LuisModel.cs]: >	 [](start = 0, length = 1)

spacing

File: CSharp/Library/Microsoft.Bot.Builder/Luis/LuisService.cs [hunk 2]
```diff
@@ -248,75 +248,86 @@ public static void Fix(LuisResult result)
...
-            return await service.QueryAsync(luisRequest, token);
-        }
-
+                }
+            }
+        }
+
+        public void ApplyThreshold(LuisResult result)
+        {
+            if (result.TopScoringIntent.Score > model.Threshold)
```
Comment (willportnoy) [inline CSharp/Library/Microsoft.Bot.Builder/Luis/LuisService.cs]: >TopScoringIntent [](start = 23, length = 16)

is this ever null?  i.e. are intents only included in Intents if verbose is true?

File: CSharp/Library/Microsoft.Bot.Builder/Luis/LuisModel.cs (hunk 1 above)
Comment (JasonSowers) [reply CSharp/Library/Microsoft.Bot.Builder/Luis/LuisModel.cs]: fixed

File: CSharp/Library/Microsoft.Bot.Builder/Luis/LuisService.cs (hunk 2 above)
Comment (JasonSowers) [reply CSharp/Library/Microsoft.Bot.Builder/Luis/LuisService.cs]: provides the same behavior with verbose true or false

File: CSharp/Library/Microsoft.Bot.Builder/Luis/LuisModel.cs [hunk 3]
```diff
@@ -117,32 +122,38 @@ public class LuisModelAttribute : Attribute, ILuisModel, ILuisOptions, IEquatabl
         /// </summary>
         public LuisApiVersion ApiVersion => apiVersion;
 
+	private readonly double threshold;
```
Comment (willportnoy) [inline CSharp/Library/Microsoft.Bot.Builder/Luis/LuisModel.cs]: >	 [](start = 0, length = 1)

spacing

**Pull request merged**